Lib/email/headerregistry.py

```diff
@@ -69,11 +69,9 @@ def addr_spec(self):
         """The addr_spec (username@domain) portion of the address, quoted
         according to RFC 5322 rules, but with no Content Transfer Encoding.
         """
-        nameset = set(self.username)
-        if len(nameset) > len(nameset-parser.DOT_ATOM_ENDS):
+        lp = self.username
+        if parser.DOT_ATOM_ENDS.intersection(lp):
```

Comment:
Consider using `set.isdisjoint()`.

Reply:
GitHub hid @serhiy-storchaka's explanation. Makes sense to stop early when finding a match via disjoint.